Format BlockClock progress percentage according to the design file

jarolrod opened the pull request:
This formats the BlockClock's progress percentage to fit with the [design file](https://www.figma.com/file/ek8w3n3upbluw5UL2lGhRx/Bitcoin-Core-App-Design?node-id=5986%3A17926&t=jywAOeHntrNKOEcG-4)

The design file shows: 
![Block clock-3](https://user-images.githubusercontent.com/23396902/221780498-aa576c90-f2b3-4a20-b88a-f03ef3c59654.png)


This PR implements 

| 0 | 0.01 | 0.1 | 1 |
| - | ---- | --- | - |
| <img width="752" alt="zero" src="https://user-images.githubusercontent.com/23396902/221780711-c8c31b20-cd5d-412b-af87-0060fb56eeec.png"> | <img width="752" alt="zero-dot-one" src="https://user-images.githubusercontent.com/23396902/221780804-ef35fa20-dd10-44ca-a22c-020404fdd775.png"> | <img width="752" alt="dot-one" src="https://user-images.githubusercontent.com/23396902/221780883-a20f865a-f22c-4a2c-a394-46537627dab9.png"> | <img width="752" alt="one" src="https://user-images.githubusercontent.com/23396902/221780949-57bb164d-dcb8-4e1c-a7b2-205b80073ddf.png"> |


[![Windows](https://img.shields.io/badge/OS-Windows-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/insecure_win_gui.zip?branch=pull/276)
[![Intel macOS](https://img.shields.io/badge/OS-Intel%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/insecure_mac_gui.zip?branch=pull/276)
[![Apple Silicon macOS](https://img.shields.io/badge/OS-Apple%20Silicon%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos_arm64/insecure_mac_arm64_gui.zip?branch=pull/276)
[![ARM64 Android](https://img.shields.io/badge/OS-Android-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/276)

johnny9 commented:
Concept ACK

johnny9 reviewed (commented):
[Screencast from 02-28-23 11:44:46.webm](https://user-images.githubusercontent.com/985648/221933695-966b7dfc-8b8a-4b77-973e-4763ae0bf334.webm)

Appears to be some initialization issues when onboarding is skipped

jarolrod commented:
@johnny9 It seems that you're actually testing https://github.com/bitcoin-core/gui-qml/pull/277? The issue you're seeing in the video is that with https://github.com/bitcoin-core/gui-qml/pull/277, we have synced set to if verificationprogress is > 99 where what it should be set to is a check on if the node is in IBD or not. What you're seeing in the video is some new headers, but we're not in IBD, so the UI gets put in a weird place. Will Update 277.

johnny9 reviewed (approved):
ACK 69589710809bb62caf7b0e605f321ec8120be191

johnny9 commented on src/qml/components/BlockClock.qml:
```diff
@@ -88,7 +88,7 @@ Item {
             PropertyChanges {
                 target: root
-                header: Math.round(nodeModel.verificationProgress * 100) + "%"
+                header:  formatProgressPercentage(nodeModel.verificationProgress * 100)
```
```suggestion
                header: formatProgressPercentage(nodeModel.verificationProgress * 100)
```

jarolrod replied:
thanks, fixed

johnny9 reviewed (changes requested):
nitpick: extra whitespace in `header` value

jarolrod commented:
Updated from [6958971 ](https://github.com/bitcoin-core/gui-qml/commit/69589710809bb62caf7b0e605f321ec8120be191)to [603d8b8](https://github.com/bitcoin-core/gui-qml/commit/603d8b85aabbf604d0d83c084d577efd9937895d), [compare](https://github.com/bitcoin-core/gui-qml/compare/69589710809bb62caf7b0e605f321ec8120be191..603d8b85aabbf604d0d83c084d577efd9937895d)

Changes: addressed review feedback

johnny9 reviewed (approved):
ACK 603d8b85aabbf604d0d83c084d577efd9937895d
